lib/active_record_shards/shard_selection.rb

```diff
@@ -85,7 +85,7 @@ def on_replica=(new_replica)
     alias_method :on_slave=, :on_replica=
 
     def options
-      { shard: @shard, slave: @on_replica, replica: @on_replica }
+      { shard: @shard, replica: @on_replica }
```

Comment:
this isn't private, which is why I left the option in there. In theory, any external consumer could call this options method and then this would be an immediate breaking change.

Reply:
I think the chances of anyone using `ShardSelection#options` directly is very slim. And it would be nice to not issue warnings because of ARS doing an internal call with deprecated terminology (which happens when calling `ConnectionSwitcher#switch_connection`)

Can you help me assess the risk of leaving this change as it is (with `:slave` key removed)? It might be enough if we just mention it as a breaking change in the (brand new) changelog.

Reply:
@craig-day this is the issue @bquorning described, overall it seems best to remove it, and make a note in Changelog.

Reply:
this allows removing the check / deprecation warning raised internally in `connection_switcher` (https://github.com/zendesk/active_record_shards/pull/237#discussion_r533829639)

Reply:
I agree that the risk is rather low for removing this, was just calling out that technically it is a breaking change. If we're OK with documenting this and removing it before we bump the next major version, I think that is reasonable.

Reply:
added this change to changelog